mlflow/deployments/base.py

```diff
@@ -126,72 +202,105 @@ def update_deployment(self, name, model_uri=None, flavor=None, config=None):
                        deployment will be updated using that flavor.
         :param config: (optional) dict containing updated target-specific configuration for the
                        deployment
+        :param endpoint_uri: URI of the Endpoint containing the deployment to update. May not be
+                             supported by all targets
         :return: None
         """
         pass
 
     @abc.abstractmethod
-    def delete_deployment(self, name):
+    def delete_deployment(self, name, endpoint_uri=None):
         """
         Delete the deployment with name ``name`` from the specified target. Deletion should be
         idempotent (i.e. deletion should not fail if retried on a non-existent deployment).
 
         :param name: Name of deployment to delete
+        :param endpoint_uri: URI of the Endpoint containing the deployment to delete. May not be
+                             supported by all targets.
         :return: None
         """
         pass
 
     @abc.abstractmethod
-    def list_deployments(self):
+    def list_deployments(self, endpoint_uri=None):
         """
         List deployments. This method is expected to return an unpaginated list of all
         deployments (an alternative would be to return a dict with a 'deployments' field
         containing the actual deployments, with plugins able to specify other fields, e.g.
         a next_page_token field, in the returned dictionary for pagination, and to accept
         a `pagination_args` argument to this method for passing pagination-related args).
 
+        :param endpoint_uri: URI of the Endpoint to list all contained deployments. May not be
+                             supported by all targets.
+
         :return: A list of dicts corresponding to deployments. Each dict is guaranteed to
                  contain a 'name' key containing the deployment name. The other fields of
                  the returned dictionary and their types may vary across deployment targets.
         """
         pass
 
     @abc.abstractmethod
-    def get_deployment(self, name):
+    def get_deployment(self, name, endpoint_uri=None):
         """
         Returns a dictionary describing the specified deployment, throwing a
         py:class:`mlflow.exception.MlflowException` if no deployment exists with the provided
-        ID.
+        name.
         The dict is guaranteed to contain an 'name' key containing the deployment name.
         The other fields of the returned dictionary and their types may vary across
         deployment targets.
 
         :param name: ID of deployment to fetch
+        :param endpoint_uri: URI of the Endpoint containing the deployment to get. May not be
+                             supported by all targets.
         """
         pass
 
     @abc.abstractmethod
-    def predict(self, deployment_name, df):
+    def predict(self, uri, df):
```

Comment:
If a subclass is created, there's no need to worry about backward-compatibility issues here:)